Add changelog guide to release process doc (#2114)
sourcegraph/handbook: content/departments/product-engineering/engineering/process/releases/index.md

Most issues are non-blocking. Fixes to non-blocking issues can be fixed in `main` by the code owner who can then `git cherry-pick` those commits into the release branch with the approval of the release captain. Alternatively, broken features can be reverted out of the release branch or disabled via feature flags if they aren't ready or are too buggy.

### CHANGELOG.md

When releasing a new version, the Release Captain may need to manually update the [CHANGELOG.md]. Follow the instruction below:

> - [ ] I understood the convention of our changelog format. Learn more from our [CHANGELOG.md].

Reviewer: ```suggestion
- [ ] Check past entries in [CHANGELOG.md] to understand the changelog format.
```

> - [ ] Create a new H2 title named `{major}.{minor}.{patch}` (the new version) after the `Unreleased` section if it doesn't already exist

Author: ```suggestion
- [ ] Create a new H2 title named `{major}.{minor}.{patch}` (the new version) after the `Unreleased` section if it doesn't already exist.
```

> - [ ] Copy all changelog entries of the commits belong to this patch release into the new H2 title `## {major}.{minor}.{patch}`, grouped into the types they were originally in (e.g. `### Added`, `### Changed`, `### Fixed`, `### Removed`).

Reviewer: Should this mention that this point is only for patch releases?
